crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1,3 +1,24 @@
+//! We have three Salsa queries for inferring types at three different granularities: scope-level,
+//! definition-level, and expression-level.
+//!
+//! Scope-level inference is for when we are actually checking a file, and need to check types for
+//! everything in that file's scopes.
```

Comment:
I'm not sure if it's worth mentioning. I'll leave it up to you. But scope level inference is also need for the expression's `HasTy` implementation